### PR TITLE
fix: certification url

### LINKED ISSUE #40867: Chinese Certificate verification URL incorrect
**Describe the bug**

The English and Espanol certifications have the correct "verify this certification at" URL, but the Chinese certification links to the `www.freecodecamp.dev` domain.

**To Reproduce**
1. Go to https://www.freecodecamp.org/certification/nhcarrigan/responsive-web-design and see the correct verify URL
2. Go to https://www.freecodecamp.org/espanol/certification/nhcarrigan/responsive-web-design and see the correct verify URL
3. Go to https://chinese.freecodecamp.org/certification/nhcarrigan/responsive-web-design and see the **incorrect** verify URL.

**Expected behavior**

The Chinese certification should correctly use `chinese.freecodecamp.org` for the verification url.

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40867 

Replaces the hardcoded URL with `window.location.origin` to capture the
protocol, subdomain, and domain dynamically.

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

### CODE REVIEW

**Comment by nhcarrigan** [comment]
Will fix the tests tomorrow, maybe.

@raisedadead do the tests run on production at all? Or only in GH Actions? Can I safely hard-code the `localhost:8000` value in the test assertion?

**Comment by ojeytonwilliams** [comment]
I think it's a good strategy to hard code values in tests, where possible. Tests want to be dumb, honestly.

I'd definitely recommend that here, since, yes, all our tests run on one machine in an action.

**Comment by ojeytonwilliams** [comment]
By the way, if this is passing locally, but failing in CI, you can push a branch to the main repo and we can see the recording.

File: cypress/integration/ShowCertification.js [hunk 1]
```diff
@@ -57,15 +57,15 @@ describe('A certification,', function() {
         .and(
           'match',
           // eslint-disable-next-line max-len
-          /https:\/\/www\.linkedin\.com\/profile\/add\?startTask=CERTIFICATION_NAME&name=Legacy Front End&organizationId=4831032&issueYear=\d\d\d\d&issueMonth=\d\d?&certUrl=https:\/\/freecodecamp\.org\/certification\/developmentuser\/legacy-front-end/
+          /'https:\/\/www\.linkedin\.com\/profile\/add\?startTask=CERTIFICATION_NAME&name=Legacy Front End&organizationId=4831032&issueYear=\d\d\d\d&issueMonth=\d\d?&certUrl=http:\/\/localhost:8000\/certification\/developmentuser\/legacy-front-end/
```
Comment (ShaunSHamilton) [inline cypress/integration/ShowCertification.js:60]: Just to mention: I am going around changing values like `localhost:8000` to `${Cypress.env('BASE_URL')}`. This way, it defaults to whatever is set in either `cypress.json` or what is set in `.gitpod.yml`

**Review by ojeytonwilliams — COMMENTED** [review]
What do you think about having multiple locations that certs can be verified at?

In principle it's fine (and this PR should do the job), but it does commit us to maintaining an ever growing collection of backlinks.

cc: @raisedadead

**Comment by nhcarrigan** [comment]
> What do you think about having multiple locations that certs can be verified at?

The only thing that changes between the links should be the subdomain (www vs chinese) and/or the language path (/espanol).

I don't see it creating too much technical debt for us, unless we change our domain or path configuration. My concern, though, is that *not* having this means a camper who earns a certificate in Chinese can only have that certificate in English.

**Comment by ojeytonwilliams** [comment]
> My concern, though, is that not having this means a camper who earns a certificate in Chinese can only have that certificate in English.

Yeah, I completely agree - that would not be good.  We need a solution and it's probably this, but it's worth pausing a sec and considering if this is okay.

For context, we should never have had had profiles at `/<username>`. That choice means every profile is initially a 404 and we're stuck with that since people link to their profiles.  I don't think having multiple cert locations is anywhere near as bad, but it is similar enough that it's worth thinking about.

**Comment by nhcarrigan** [comment]
True - and this approach would cause errors later on, if we (as an example) decided to host languages on subdomains instead of subdirectories.

**Comment by nhcarrigan** [comment]
Closing per discussion in this morning's meeting - will leave branch in-tact until we determine approach 🙂

**Pull request closed without merging**